pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -164,9 +164,32 @@ func commonArgs(cluster string, cfg *config.Cluster) ([]string, error) {
 	if usernsRemap() {
 		args = append(args, "--userns=host")
 	}
+
+	// handle Docker on Btrfs or ZFS
+	dockerStorage, err := getDockerStorage()
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get Docker Storage")
+	}
+	if dockerStorage == "btrfs" || dockerStorage == "zfs" {
+		args = append(args, "--volume", "/dev/mapper", "/dev/mapper")
```

Comment:
implementation wise - LGTM.
i'd add a note here linking to external documentation why `/dev/mapper` solves the issue (if it indeed does).